filter/filter_test.go

```diff
@@ -218,6 +238,38 @@ func TestFilters(t *testing.T) {
 		{`s = "Buenos d\u{ed}as se\u{f1}or"`, true},
 	})
 
+	// There are two Unicode code points with a multibyte UTF-8 encoding
+	// equivalent under Unicode simple case folding to code points with
+	// single-byte UTF-8 encodings: U+017F LATIN SMALL LETTER LONG S is
+	// equivalent to S and s, and U+212A KELVIN SIGN is equivalent to K and
+	// k. The next two records ensure they're handled correctly.
+
+	// Test U+017F LATIN SMALL LETTER LONG S.
+	tzng = `
+#0:record[a:string]
+0:[\u017F;]`
+	runCases(t, tzng, []testcase{
+		{`a = \u017F`, true},
+		{`a = S`, false},
+		{`a = s`, false},
+		{`\u017F`, true},
+		{`S`, false}, // XXX should be true
+		{`s`, false}, // XXX should be true
```

Comment:
That these are false reflects bugs in both filter.stringSearch and filter.stringCaseFinder.next, for which I'll create an issue.

Reply:
#1207 tracks this.